### ISSUE TITLE
Styles repeated in `vendor` and `engine-vendor` output

### ISSUE THREAD

**Issue by alexlafroscia** [salsify/ember-css-modules#127]
Our application has a main, "root" application that depends on an internal addon of shared components, and also mounts a few lazy engines that _also_ depend on that addon. The graph is sort of like this:

```
core-app
|-- shared-components
|-- lazy-engine
  |-- shared-components
```

I've noticed that both `vendor.css` _and_ `engine-vendor.css` include the stylesheet for the `shared-components` addon

I'm not sure if it's possible, but it would be really cool if `ember-css-modules` was "smart enough" to know that both the engine _and_ the host application have the addon installed and avoid including the styles a second time in the `engine-vendor.css` file.

Note: it doesn't have a negative impact on the application (other than the cost of the browser processing CSS that is just overwritten by the exact same class definition in another file) but it would be nice to avoid the payload cost of repeating the same content in two different files.

**Comment by dfreeman** [comment]
Agreed that this is far from ideal, but I'm not sure if there's a trivial way to fix this in the short term. Right now I'm very interested to see if the [embroider](https://github.com/embroider-build/embroider) project (which is a proposal for a v2 version of the build pipeline) gains traction, because it could pretty dramatically change how much of this needs to work.
